### DISCUSSION TITLE
[ERROR DRT-0073] No access point for foobar

### DISCUSSION THREAD

**Discussion by oharboe** [The-OpenROAD-Project/OpenROAD#3181] [General]
What does this mean?

"foobar" is the first iterm of the macro and the macro is very tall and has lots of wires on the left hand side.

![image](https://user-images.githubusercontent.com/2798822/232051624-f59d4343-6a83-4e02-9e7b-f4518eb69bbf.png)

**Comment by maliberty** [comment]
At the start of drt we figure out how we can access each pin we need to connect to.  The access point is a spot where we could legally drop a via or make a planar connection (the later is more common for macro pins).  This error results when we can't find any legal way to connect to the pin so the router gives up early rather than struggle on hopelessly.

The analysis is quite local to the pin and its immediate surrounding fixed objects (i.e. other pins or blockages).  It would be helpful to have more of a close up image of the pin to get a better sense of what is going on.

**Reply by oharboe** [reply]
Fixed: https://github.com/The-OpenROAD-Project/OpenROAD/pull/3196
